### PR TITLE
[Feat] Support WKB geometry column in CSV

### PR DESCRIPTION
**Pull request by lixun910**
In Kepler.gl, geometries (Polygons, Points, LindStrings etc) can be embedded into CSV as a GeoJSON or WKT formatted string (see doc [here](https://docs.kepler.gl/docs/user-guides/b-kepler-gl-workflow/a-add-data-to-the-map#3.-embed-geometries-in-csv)). This PR is to support WKB geometry column in CSV file. 

For example, the following CSV row contains a WKT geometry:
```
id,geom
2,POINT(1 2)
```

the following CSV row contains a WKB geometry in hex format:
```
id, geom
1,0101000000000000000000F03F0000000000000040
```
(In PostGIS, the WKB in hex is the default format of the output of the geometry column, unless using e.g. ST_AsWKT)

1. The WKB format is more efficient than the WKT for processing geometry objects (no parsing needed).
3. The WKB format keeps the precisions of the raw geometry data
4. The size of WKB in hex format (2 times the size of WKB) is usually smaller than WKT with complex geometries (linestrings, polygons etc. -- not for simple points)

- guerry.geojson is 2 MB
- guerry_wkt.csv is 1.7 MB 
- guerry_wkb.csv is 1.4 MB
- guerry.arrow is 705 KB

For Arrow/GeoArrow, there is another wip PR to support Arrow/GeoArrow in Kepler.gl: https://github.com/GeoDaCenter/kepler.gl/pull/2

### CODE REVIEW

File: src/layers/src/geojson-layer/geojson-utils.ts [hunk 1]
```diff
@@ -138,6 +141,17 @@ export function parseGeometryFromString(geoString: string): Feature | null {
     }
   }
 
+  // try parse as wkb using loaders.gl WKBLoader
+  if (!parsedGeo) {
+    try {
+      const buffer = Buffer.from(geoString, 'hex');
+      const binaryGeo = parseSync(buffer, WKBLoader);
```
Comment (ibgreen) [inline src/layers/src/geojson-layer/geojson-utils.ts:148]: This looks great. We should probably just use the `WKTLoader` from loaders.gl/wkt for the wkt parsing on line 138. It is based on wellknown so it should be equivalent but keeping consistent API.

File: src/layers/src/geojson-layer/geojson-utils.ts [hunk 2]
```diff
@@ -138,6 +141,17 @@ export function parseGeometryFromString(geoString: string): Feature | null {
     }
   }
 
+  // try parse as wkb using loaders.gl WKBLoader
+  if (!parsedGeo) {
+    try {
+      const buffer = Buffer.from(geoString, 'hex');
```
Comment (ibgreen) [inline src/layers/src/geojson-layer/geojson-utils.ts:147]: Nit: Wish we had a loader for this so we didn't need to mess with Node.js Buffer class. I will give it some thought.

File: src/utils/src/dataset-utils.ts [hunk 3]
```diff
@@ -388,6 +388,27 @@ export function getSampleForTypeAnalyze({
...
 }
 
+/**
+ * Check if string is a valid Well-known binary (WKB) in HEX format
+ * https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry
+ *
+ * @param str input string
+ * @returns true if string is a valid WKB in HEX format
+ */
+export function isHexWkb(str: string | null): boolean {
```
Comment (ibgreen) [inline src/utils/src/dataset-utils.ts:398]: This is interesting. I wonder how to handle HexWKB in loaders.gl. Maybe just add `HexWKBLoader` / `HexWKBWriter`?

**Review by ibgreen — APPROVED** [review]
Looks great, thanks for contribution!

File: tsconfig.json [hunk 4]
```diff
@@ -21,7 +21,7 @@
     "strict": true,
     "resolveJsonModule": true,
     "isolatedModules": true,
-    "baseUrl": "./src",
+    "baseUrl": ".",
```
Comment (ibgreen) [inline tsconfig.json:24]: What does this do? Looks unrelated?

File: src/layers/src/geojson-layer/geojson-utils.ts (hunk 1 above)
Comment (lixun910) [reply src/layers/src/geojson-layer/geojson-utils.ts:148]: Agreed. Will try to use WKTLoader instead of wktParser from wellknown package. Thanks!

File: src/utils/src/dataset-utils.ts (hunk 3 above)
Comment (lixun910) [reply src/utils/src/dataset-utils.ts:398]: HexWKBLoader/Writer sounds great 👍

**Review by lixun910 — COMMENTED** [review]
Thanks for your comments, @ibgreen!

File: tsconfig.json (hunk 4 above)
Comment (lixun910) [reply tsconfig.json:24]: This is not related to WKB, but it seems not been set correctly and causing the alias "@kepler.gl/*" not mapping to correct source code. (see below `@kepler.gl/*": ["src/*/src"]`)

File: src/layers/src/geojson-layer/geojson-utils.ts (hunk 2 above)
Comment (lixun910) [reply src/layers/src/geojson-layer/geojson-utils.ts:146]: Thanks! Ib will address this in https://github.com/visgl/loaders.gl/pull/2652

**Comment by lixun910** [comment]
@ibgreen added new functions e.g. `HexWKBLoader` that can be used to directly parse the HexWKB column using loaders.gl. See PR [here](https://github.com/visgl/loaders.gl/pull/2652). Thanks, @ibgreen! He also plans to backport this PR to 3.4 and make a patch. 

Let me know if merging this PR is OK, or if we should wait for the patch in loaders.gl v3. Thanks!

**Comment by ibgreen** [comment]
Yes its fine to merge this now. It will take a while to issue a loaders 3.4 patch.
